### PR TITLE
Problem: nixpkgs is not pinned properly

### PR DESCRIPTION
**Pull request by yihuang**
nixpkgs is only pinned with environment variable, better to use niv to manage.

Solution:
- use niv to manage nixpkgs and other dependencies
- refactor nix expression a little bit

### CODE REVIEW

**Comment by tomtau** [comment]
the ledger test fails @linfeng-crypto ?
```
==================================== ERRORS ====================================
4175
____________________ ERROR at setup of test_ledger_transfer ____________________
4176

4177
pytestconfig = <_pytest.config.Config object at 0x7fee73a14fd0>
4178
tmp_path_factory = TempPathFactory(_given_basetemp=None, _trace=<pluggy._tracing.TagTracerSub object at 0x7fee73838850>, _basetemp=PosixPath('/tmp/pytest-of-runner/pytest-1'))
4179

4180
    @pytest.fixture(scope="module")
4181
    def cluster(pytestconfig, tmp_path_factory):
4182
        "override cluster fixture for this test module"
4183
        ledger = get_ledger()
4184
        ledger.start()
4185
        try:
4186
>           yield from cluster_fixture(
4187
                Path(__file__).parent / "configs/ledger.yaml",
4188
                26800,
4189
                tmp_path_factory,
4190
                quiet=pytestconfig.getoption("supervisord-quiet"),
4191
            )
4192

4193
integration_tests/test_hardware_wallet.py:16: 
```

**Comment by yihuang** [comment]
> the ledger test fails @linfeng-crypto ?
> 

I think it's my fault, it's running with the chain-maind without ledger-zemu.

File: Makefile [hunk 1]
```diff
@@ -146,8 +146,8 @@ clean-docker-compose: localnet-stop
 ###############################################################################
 # nix installation: https://nixos.org/download.html
 nix-integration-test: check-network
-	nix-shell integration_tests/shell.nix --arg ci true --run "pytest -v -n 6 -m 'not ledger' --dist loadscope"
-	nix-shell integration_tests/shell.nix --arg ci true --run "pytest -v -m ledger"
+	nix-shell ./. -A ci-shell --run "pytest -v -n 7 -m 'not ledger' --dist loadscope"
```
Comment (leejw51crypto) [inline Makefile:149]: what's meaning of ./.?

File: .github/workflows/nix.yml [hunk 2]
```diff
@@ -14,31 +14,18 @@ jobs:
...
-      env:
-        SHELL: >
-          with import <nixpkgs> {};
-          mkShell {
-            buildInputs = [
-              (poetry2nix.mkPoetryEnv { projectDir = ./integration_tests; })
-              nixpkgs-fmt
-            ];
-          }
+      run: nix-shell ./. -A lint-shell --run "make lint-ci"
```
Comment (leejw51crypto) [inline .github/workflows/nix.yml:22]: how about nix-shell $PWD?

**Review by leejw51crypto — APPROVED** [review]
lgtm

File: Makefile (hunk 1 above)
Comment (yihuang) [reply Makefile:149]: it's short for `./default.nix`, I guess you can also use `$PWD/.`.

**Review by tomtau — APPROVED** [review]

**Pull request merged**